src/iso15118/session/iso.cpp

```diff
@@ -157,14 +157,18 @@ TimePoint const& Session::poll() {
     }
 
     // send all of our queued control events
-    while (active_control_event = control_event_queue.pop()) {
+    active_control_event =
+        control_event_queue.pop(); // TODO(sl): Not sure, -Wparentheses does not like the previous solution
```

Comment:
To assign with a conditional check, the assignment can be put into additional brackets:
```
while ((active_control_event = control_event_queue.pop())) {
```

While it looks like a hack to shut up the warnings, it's apparently the way to tell the compiler about your intent.

Reply:
https://stackoverflow.com/questions/24116817/when-do-extra-parentheses-have-an-effect-other-than-on-operator-precedence/24116818#24116818

additional parentheses can have a real subtle effect

Reply:
but here, they are used to express intend

Reply:
Done